Azure-Samples/communication-services-android-calling-hero
# Add feature to select audio output device

- Add audio device select menu on both SetupActivity & CallActivity.
- With the menu, user can select earpiece "Android" or Speaker as the audio output.
- Android selection also includes the other type of external devices (Headset, Bluetooth...)

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Test the code

- Pull the code base from "audio-device" branch
- Add necessary appSettings info
- Start the App
- At the call setup screen, there will be one more button next to "Mic" button called "Device"
- Press "Device" button, you will see audio device menu. Then you can choose either Android/Earpiece or Speaker as audio output.
- You can tap the area outside the device menu to close the menu
- At the call activity screen, there will be one more button added left-behind the "red button" which is also the "Device" button.
- You can choose audio output at any time during the call. 
- Rotate the screen during the call will keep the invisibility of the device menu. You can touch area outside the device menu to close the nemu.
